### DISCUSSION TITLE
PySide6 + pyqtgraph recurssion error

### DISCUSSION THREAD

**Discussion by ttortorici** [pyqtgraph/pyqtgraph#2393] [Q&A]
____________
Edit: it is `from __feature__ import snake_case` causing the issue.
_____________

I am having trouble implementing pyqtgraph with PySide6. I have used pyqtgraph with PyQt5 without a hitch, but I'm making a new application in PySide6 specifically for a new project.

I checked the pyqtgraph documentation and it says to import your Qt wrapper before pyqtgraph so it knows which to work with.

```
import sys
from PySide6.QtWidgets import QMainWindow, QApplication, QTabWidget

from gui_scripts.tab_plot import PlotTab
import PySide6
import pyqtgraph as pg

from __feature__ import snake_case, true_property


class MainWindow(QMainWindow):
    def __init__(self):
        QMainWindow.__init__(self)
        self.graph_widget = pg.PlotWidget()
        self.set_central_widget(self.graph_widget)
        hour = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
        temperature = [30, 32, 34, 32, 33, 31, 29, 32, 35, 45]
        self.graph_widget.plot(hour, temperature)


if __name__ == "__main__":
    app = QApplication(sys.argv)

    main_window = MainWindow()
    main_window.show()

    sys.exit(app.exec())
```

I get a "RecursionError: maximum recursion depth exceeded while calling a Python object" triggered where self.graph_widget = pg.PlotWidget()

I used this example https://www.pythonguis.com/tutorials/pyside-plotting-pyqtgraph/ to generate the same error because my actual implementation is across a few scripts. The application has a navigation tab that switches between a few interfaces, one of which will be a plot, and the code worked just fine until I added the tab for the plot.

When I remove snake_case, and change the method names to camelCase, it makes no difference.

**Comment by j9ac9k** [comment]
Hi @teddy-tort 

Thanks for diagnosing the issue!  the `from __future__ import snake_case, true_property` is not a feature I personally am unlikely to take up.  I'll certainly review a PR that makes it so that functionality works, but I doubt there will be much interest from the current maintainers to take this on.  I haven't thought of this thoroughly, but at first thought, this would likely break almost everything in our [Qt abstraction layer](https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/Qt/__init__.py).
